Dolphin scripts/Entrance Randomizer/lib/utils.py

```diff
@@ -92,17 +92,88 @@ def follow_pointer_path(ppath: Sequence[int]):
     return addr
 
 
+def prevent_item_softlock():
+    """
+    Prevent softlocking by missing the right items.
+
+    Even with logic this can happen if the player bypasses logic.
+    """
+    # TODO: A better fix would be to just teleport the player to the entrance of the temple,
+    # but we can't reliably do that yet.
+    # And we can't use position-based fix as the game checks for walls even when teleporting
+
+    # Pickaxe lets you escape everything
+    if memory.read_u32(ADDRESSES.backpack_struct + BackpackOffset.Pickaxes):
+        return
+
+    # Scorpion Temple
+    if (
+        not memory.read_u32(ADDRESSES.backpack_struct + BackpackOffset.Torch)
+        and state.current_area_old == LevelCRC.SCORPION_SPIRIT
+        and state.current_area_new == LevelCRC.SCORPION_TEMPLE
+    ):
+        # Lets just give the player the torch, it's not like it unlocks much outside convenience.
```

Comment:
If you're going to give Torch in Scorpion Temple, it would be appropiate to also add Scorpion Temple to the `UPGRADE_AREAS` in `graph_creation.py`.

Reply:
Whilst it's made to be temporary, I see your point. Hopefully players don't get too used to it.